src/libasr/pickle.cpp

```diff
@@ -52,13 +52,13 @@ class ASRPickleVisitor :
     }
     void visit_Module(const ASR::Module_t &x) {
         if (!show_intrinsic_modules &&
-                    startswith(x.m_name, "lfortran_intrinsic_")) {
+            (x.m_intrinsic || startswith(x.m_name, "numpy"))) { // Divergence between LFortran and LPython
```

Comment:
Here let's keep both?

Reply:
We can do that. I am unsure, but it may/might change the reference tests. The idea in this PR is to remove redundant/unused code, ideally keeping the other things intact. Hence, the above divergence comment.

I think it can be discussed/tackled later, not a priority at the moment.

Reply:
But still, let's fix this the right way.
We will keep both checks; if it encounters any of the intrinsic module names, then it is marked as a IntrinsicModule.
So I would do this as:
```suggestion
            (x.m_intrinsic || startswith(x.m_name, "lfortran_intrinsic_") || startswith(x.m_name, "numpy"))) { // Divergence between LFortran and LPython
```

Reply:
This would be synced with LFortran Libasr in the next sync.

Reply:
Let me see if it changes the reference tests. If the reference tests change, I would prefer to do it in a separate PR.

Reply:
The reference tests didn't change, so I think it is fine.